controllers/storagecluster/generate.go

```diff
@@ -79,7 +80,22 @@ func generateNameForSnapshotClassDriver(initData *ocsv1.StorageCluster, snapshot
 	return fmt.Sprintf("%s.%s.csi.ceph.com", initData.Namespace, snapshotType)
 }
 
-func generateNameForSnapshotClassSecret(snapshotType SnapshotterType) string {
+func generateNameForSnapshotClassSecret(instance *ocsv1.StorageCluster, snapshotType SnapshotterType) string {
+	if instance.Spec.ExternalStorage.Enable == true {
+		data, ok := externalOCSResources[instance.UID]
```

Comment:
It's the json data from the python script, initialised by ocs-operator

Reply:
AFAIK in consumer mode, we dont call this function nor newVolumeSnapshotClass function also.

Reply:
We do call it https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/reconcile.go#L405

Reply:
it will move skipped for consumer mode here https://github.com/red-hat-storage/ocs-operator/pull/1729/files#diff-ebb1f458b4aa3b51172933994b017c626698cd74c8de4b9fcf63ae92f5edc290R133